game-core/src/main/java/games/strategy/engine/posted/game/pbf/NodeBbTokenGenerator.java

```diff
@@ -0,0 +1,129 @@
+package games.strategy.engine.posted.game.pbf;
+
+import com.google.common.base.Preconditions;
+import games.strategy.engine.framework.system.HttpProxy;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+import java.util.Map;
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import lombok.AccessLevel;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import org.apache.http.client.entity.UrlEncodedFormEntity;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpDelete;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.http.message.BasicNameValuePair;
+import org.apache.http.util.EntityUtils;
+import org.snakeyaml.engine.v2.api.Load;
+import org.snakeyaml.engine.v2.api.LoadSettings;
+
+@AllArgsConstructor
+public class NodeBbTokenGenerator {
+  private final String forumUrl;
+  private final Load load = new Load(LoadSettings.builder().build());
+
+  @AllArgsConstructor(access = AccessLevel.PRIVATE)
+  @Getter
+  public static class TokenInfo {
+    @Nonnull private final String token;
+    private final int userId;
+  }
+
+  public TokenInfo generateToken(
+      final String username, final String password, @Nullable final String otp) {
+    Preconditions.checkNotNull(username);
+    Preconditions.checkNotNull(password);
+
+    try (CloseableHttpClient client = HttpClients.custom().disableCookieManagement().build()) {
+      final int userId = getUserId(client, username);
+      return new TokenInfo(getToken(client, userId, password, otp), userId);
+    } catch (final IOException e) {
+      throw new RuntimeException("Failed to retrieve login token", e);
+    }
+  }
+
+  public void revokeToken(final String token, final int userId) {
+    try (CloseableHttpClient client = HttpClients.custom().disableCookieManagement().build()) {
+      deleteToken(client, userId, token);
+    } catch (final IOException e) {
+      throw new RuntimeException("Failed to revoke login token", e);
+    }
+  }
+
+  private void deleteToken(final CloseableHttpClient client, final int userId, final String token)
+      throws IOException {
+    final HttpDelete httpDelete =
+        new HttpDelete(forumUrl + "/api/v2/users/" + userId + "/tokens/" + token);
```

Comment:
Any thoughts to first, or as part of this PR, to convert the http client to Feign?

I mention that for a few reasons:
- the different styles/technologies for http communication is inconsistent. It's one of the technology consistencies that a project really should try to keep so you limit how many technologies you need to learn and how many different patterns there are for doing something.
- the test infrastructure for http communication with feign client is pretty straight forward. At this point with wiremock, we should be testing this client, going to a consistent pattern and using the same technological infrastructure would make that easy.

Reply:
I was thinking about migrating away from http component to either use the built-in java API that's pretty neat or feign, but not as part of this PR